src/control/lib/control/rpc.go

```diff
@@ -392,5 +392,7 @@ func invokeUnaryRPC(parentCtx context.Context, log debugLogger, c UnaryInvoker,
 // items which represent the success or failure of the RPC invocation for each host
 // in the request.
 func (c *Client) InvokeUnaryRPC(ctx context.Context, req UnaryRequest) (*UnaryResponse, error) {
+	req.SetSystem(c.config.SystemName)
```

Comment:
I think you need to check to see if the request has a system set already before forcing it. It's true that we're removing the system from the dmg command surface for now, but there's no good reason not to make this future-proof as far as I can see.

Reply:
Rethinking this it doesn't make sense to set system name here, the canonical value needs to be evaluated before setting in protobuf request before passing the request to MgmtSvcClient method in setRPC call. I have refactored to retrieve any configured value from the Client (UnaryInvoker) in request.getSystem(), Let me know what you think.